recovery/install/src/Requirements.php

```
@@ -217,6 +224,44 @@ private function checkModRewrite()
         return isset($_SERVER['MOD_REWRITE']);
     }
 
+    /**
+     * Checks the opcache configuration if the opcache exists.
+     */
+    private function checkOpcache()
+    {
+        if (!extension_loaded('Zend OPcache')) {
+            return [];
+        }
+
+        $useCwdOption = $this->compare('opcache.use_cwd', ini_get('opcache.use_cwd'), 1);
+        $opcacheRequirements = [[
+            'name' => 'opcache.use_cwd',
+            'group' => 'core',
+            'required' => 1,
+            'version' => ini_get('opcache.use_cwd'),
+            'result' => ini_get('opcache.use_cwd'),
+            'notice' => '',
+            'check' => $this->compare('opcache.use_cwd', ini_get('opcache.use_cwd'), 1),
+            'error' => '',
+        ]];
+
+        if (fileinode('/') > 2) {
```

This is very magic, what does this check do ? This seems to cause problems with openbasedir restrictions on some systems. is this necessary? see also https://issues.shopware.com/issues/NEXT-10564

---

You should consider using chrooted php-fpm. open_basedir introduces many performance issues cause of disabled realpath cache

---

Remark: I don't use open basedir, just tried try help a bit in the linked issue.

Still trying to understand this if statement out of curiosity.

And if open basedir is officially not supported it should go to the readme

---

We did modified the patch after import and had the same problem. But it should run into this empty catch https://github.com/shopware/shopware/blob/406925fbb6a8e98beb94e71ceca1fd473a6edc31/recovery/install/src/Requirements.php#L256

---

This has been wrong copied to Shopware 6. So I don't see an issue here :)